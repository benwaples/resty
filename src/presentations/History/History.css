#History {
  list-style-type: none;
}