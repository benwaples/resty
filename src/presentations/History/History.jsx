import React from 'react';
import PropTypes from 'prop-types';
import Request from './Request/Request';
import styles from './History.css';



function History({ history, onClick }) {
  

  // eslint-disable-next-line max-len
  const previousRequests = history.map(request => <Request onClick={onClick} method={request.method} url={request.url} key={`${request.url}-${request.method}-${new Date()}`}/>);

  return (
    <ul id={styles.History}>
      {previousRequests}
    </ul>
  );
}

History.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired,
  onClick: PropTypes.func.isRequired
};

export default History;

