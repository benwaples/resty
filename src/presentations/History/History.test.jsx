import React from 'react';
import { cleanup, render } from '@testing-library/react';
import History from './History';

describe('Old request component', () => {
  afterEach(() => cleanup());

  it('should render an old request', () => {
    const history = [
      {
        url: 'google.com',
        method: 'GET'
      },
      {
        url: 'yahoo.com',
        method: 'GET'
      },
      {
        url: 'wiseguy.com',
        method: 'GET'
      },
    ];
    const { asFragment } = render(<History history={history}/>);

    expect(asFragment()).toMatchSnapshot();
  });
});
