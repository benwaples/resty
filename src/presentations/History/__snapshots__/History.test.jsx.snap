// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Old request component should render an old request 1`] = `
<DocumentFragment>
  <ul
    id="History"
  >
    <li
      class="Request"
    >
      <p
        class="method"
      >
        GET
      </p>
      <p
        class="url"
      >
        google.com
      </p>
    </li>
    <li
      class="Request"
    >
      <p
        class="method"
      >
        GET
      </p>
      <p
        class="url"
      >
        yahoo.com
      </p>
    </li>
    <li
      class="Request"
    >
      <p
        class="method"
      >
        GET
      </p>
      <p
        class="url"
      >
        wiseguy.com
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
